public/js/data.js

```diff
@@ -240,17 +213,78 @@ function checkIfBackup(scope) {
 
 // Function that should backup if there is a change, otherwise ignore
 function scheduleBackup(scope = globalScope) {
-
     var backup = JSON.stringify(backUp(scope));
     if (scope.backups.length == 0 || scope.backups[scope.backups.length - 1] != backup) {
         scope.backups.push(backup);
         scope.timeStamp = new Date().getTime();
         projectSaved = false;
     }
-
     return backup;
 }
 
+const undoRedo = {
```

Comment:
yeah, I decided not to, for easy readability.

Reply:
no way, duplicating it will not make it readable at all.
but any way GOOD JOB 👍 

![image](https://user-images.githubusercontent.com/41790090/80307332-b51ce480-87c8-11ea-8f74-c1803c2841c2.png)